src/files/handler.js

```diff
@@ -7,40 +8,45 @@ init();
  * Firebase Cloud Function handler for SvelteKit
  *
  * This function converts the Firebase Cloud Function (Express.js) Request object
- * into a format consumable to the SvelteKit render() function, which is of type
- * SvelteKit `import('types/hooks').StrictBody | null`
+ * into a format consumable to the SvelteKit render() function
  *
  * Relevant documentation - https://firebase.google.com/docs/functions/http-events#read_values_from_the_request
  *
  * @param {import('firebase-functions').https.Request} request
  * @param {import('express').Response} response
  * @returns {Promise<void>}
  */
-const svelteKit = async ({body, headers, method, rawBody, url}, response) => {
-	const host = `${headers['x-forwarded-proto']}://${headers.host}`;
-	const {pathname, searchParams: searchParameters = ''} = new URL(url || '', host);
+async function svelteKit(request, response) {
+	const rendered = await render(toSvelteKitRequest(request));
 
-	const finalRawBody =
-		headers['content-type'] === undefined ?
-			rawBody :
-			(headers['content-type'] === 'application/octet-stream' ?
-				body :
-				new TextDecoder(headers['content-encoding'] || 'utf-8').decode(rawBody));
+	return rendered ?
+		response.writeHead(rendered.status, rendered.headers).end(rendered.body) :
+		response.writeHead(404, 'Not Found').end();
+}
 
-	const rendered = await render({
-		method,
-		headers,
-		path: pathname,
-		query: searchParameters,
-		rawBody: finalRawBody
-	});
-
-	if (rendered) {
-		const {status, headers, body} = rendered;
-		return response.writeHead(status, headers).end(body);
-	}
+/**
+ * @param {import('firebase-functions').https.Request} request
+ * @return {import('@sveltejs/kit/types/internal').Incoming}
+ */
+function toSvelteKitRequest(request) {
+	const host = `${request.headers['x-forwarded-proto']}://${request.headers.host}`;
+	const {pathname, searchParams: searchParameters} = new URL(
+		request.url || '',
+		host
+	);
 
-	return response.writeHead(404).end();
-};
+	return {
+		// Incoming
+		method: request.method,
+		headers: request.headers,
+		rawBody: new Uint8Array(request.rawBody),
+		// Body: request.body, Why does SvelteKit not use body?
```

Comment:
SvelteKit types say there is also an optional `body:ParameterizedBody` field. Not sure what this is for or if we should populate it.

Reply:
I would guess that SvelteKit populates it from `rawBody` and that the adapters don't need to though I haven't looked into it

Reply:
`params` will be removed, it was a holdover from when I was figuring out the available fields and types (Incoming & Location) obvs I later found the final type which omits params.